firebase-common/src/main/java/com/google/firebase/emulators/FirebaseEmulators.java

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.emulators;
+
+import com.google.android.gms.common.annotation.KeepForSdk;
+
+/**
+ * Enum for Firebase services that can be emulated using the Firebase Emulator Suite.
+ *
+ * <p>TODO(samstern): Un-hide this once Firestore, Database, and Functions are implemented
+ *
+ * @see com.google.firebase.FirebaseApp#enableEmulators(EmulatorSettings)
+ * @see EmulatorSettings
+ * @see EmulatedServiceSettings
+ * @hide
+ */
+@KeepForSdk
+public enum FirebaseEmulators {
```

Comment:
@vkryachko if these were string constants rather than an enum would that help?  Then I don't think all BaaS would have to be released after each emulator is released, only `common` and the SDK which corresponds to the new emulator.

Reply:
@samtstern imo it would, although additionally - as I mentioned below - ideally those constants should not live inside common either but rather in their respective sdks.

Reply:
Ah ok your suggestion below is a good one!